forms-flow-api/src/formsflow_api/resources/cmis_connector.py

```diff
@@ -0,0 +1,103 @@
+from flask_restx import Namespace, Resource
+
+from formsflow_api.utils import auth, cors_preflight, profiletime
+from formsflow_api.services.external.cmislib.atompub.binding import AtomPubBinding
+
+# keeping the base path same for cmis operations (upload / download) as cmis/
+
+API = Namespace("CMIS", description="CMIS Connector")
+
+
+@cors_preflight("GET,POST,OPTIONS")
+@API.route("/upload", methods=["POST", "OPTIONS"])
+class CMISConnectorUploadResource(Resource):
+    """Resource for uploading cms repo."""
+
+    @staticmethod
+    @auth.require
+    @profiletime
+    def post():
+        """New entry in cms repo with the new resource."""
+        cms_repo_url = current_app.config.get("CMS_REPO_URL")
+        cms_repo_username = current_app.config.get("CMS_REPO_USERNAME")
+        cms_repo_password = current_app.config.get("CMS_REPO_PASSWORD")
+        if cms_repo_url is None:
+            return {"message": "CMS Repo Url is not configured"}, HTTPStatus.INTERNAL_SERVER_ERROR
+
+        args = {'binding': AtomPubBinding()}
+        client = CmisClient(cms_repo_url, cms_repo_username, cms_repo_password, **args)
+        repo = client.defaultRepository
+        results = repo.query("select * from cmis:folder where cmis:name = 'uploads'")
+        if len(results) == 0:
+            root_folder = repo.rootFolder
+            uploads = root_folder.createFolder("uploads")
+        else:
+            uploads = repo.getObjectByPath("/uploads")
+
+        if 'upload' not in request.files:
+            return {"message": "No upload files in the request"}, HTTPStatus.BAD_REQUEST
+
+        contentfile = request.files['upload']
+        filename = contentfile.filename
+        if filename != '':
+            try:
+                document = uploads.createDocument(filename, contentFile=contentfile)
+                return (
+                    (
+                        {
+                            "objectId": document.getObjectId(),
+                            "name": document.name,
+                        }
+                    ),
+                    HTTPStatus.OK,
+                )
+            except UpdateConflictException:
+                return {"message": "The uploaded file already existing in the repository"}, \
+                       HTTPStatus.INTERNAL_SERVER_ERROR
+        else:
+            return {"message": "No upload files in the request"}, HTTPStatus.BAD_REQUEST
+
+
+@cors_preflight("GET,POST,OPTIONS")
+@API.route("/download", methods=["GET", "OPTIONS"])
+class CMISConnectorDownloadResource(Resource):
+    """Resource for downloading files from cms repo."""
+
+    @staticmethod
+    @auth.require
+    @profiletime
+    def get():
+        """Getting resource from cms repo."""
+        cms_repo_url = current_app.config.get("CMS_REPO_URL")
+        cms_repo_username = current_app.config.get("CMS_REPO_USERNAME")
+        cms_repo_password = current_app.config.get("CMS_REPO_PASSWORD")
+        if cms_repo_url is None:
```

Comment:
🐛 This will cause a bug. since the default value for `cms_repo_url` is an empty string this condition won't match.
@sumesh-aot 
```suggestion
        if not cms_repo_url:
```

Reply:
This is take care at the config level